lib/pubid/ieee/renderer/base.rb

```diff
@@ -77,7 +77,7 @@ def render_edition(edition, _opts, _params)
       if edition[:month]
         month = edition[:month]
         month = Date.parse(edition[:month]).month if month.to_i.zero?
-        result += "-#{sprintf('%02d', month)}"
+        result += "-#{sprintf('%02d', month.to_i)}"
```

Comment:
Can you move the type cast into integer into the parsing instead of rendering? Same for all date components. Thanks.

Reply:
@ronaldtse I moved conversion code to Identifiers::Base. Please check if it's acceptable.